### PR TITLE
[Gekidou MM-44209] Markdown heading style updates

### PR DESCRIPTION
**Pull request by matthewbirtch**
#### Summary
Udpates heading styles to use Metropolis and adjust some spacing for markdown rendering.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-44209

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: iOS Simulator (iPhone 13), Android Simulator (Pixel 3)

#### Screenshots
<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
<tr>
<td>
image
</td>
<td>
image
</td>

```release-note
NONE
```

### CODE REVIEW

**Review by shazm — APPROVED** [review]
Changes look good to me - I do wonder if it's worth swapping over and using the typography helper to keep all the text consistent?

https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/utils/typography.ts; use via `heading1: { ...typography('Heading', 500) }`

**Comment by enahum** [comment]
👍  on the typography helper.. (1/5) really..

@matthewbirtch can you make sure this is also working as expected on Android? I see you only tested iOS

**Comment by matthewbirtch** [comment]
> Changes look good to me - I do wonder if it's worth swapping over and using the typography helper to keep all the text consistent?
> 
> https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/utils/typography.ts; use via `heading1: { ...typography('Heading', 500) }`

Great idea @shazm. I'll give that a try tomorrow and will see if I can sort that out. Definitely makes sense to keep it consistent.

> @matthewbirtch can you make sure this is also working as expected on Android? I see you only tested iOS

Sure thing @enahum. I should have marked this as a Draft PR as I was going to review in Android as well before requesting any reviews.

**Comment by matthewbirtch** [comment]
@shazm @enahum typography is now being used. Thanks for the suggestion!

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
Waiting on the Android review before merging

**Comment by matthewbirtch** [comment]
Android simulator review looks good @enahum

**Pull request merged**
